### DISCUSSION TITLE
Kodi add-on for Philips Hue

### DISCUSSION THREAD

**Discussion by oleksandr-belei** [hass-emulated-hue/core#92] [Ideas]
Can we hope for support for this plugin in future releases? 
[https://github.com/zim514/script.service.hue](https://github.com/zim514/script.service.hue)

**Comment by alexyao2015** [comment]
Is it not working now and if not, what is not working?

**Reply by oleksandr-belei** [reply]
I can't get it to work... After creating a pair with Emulated Hue, I see an error in the logs:

`2021-01-24 22:04:11,316 WARNING emulated_hue.api -- Invalid/unknown request: <Request GET /api/dab3bbc2-a4e0-4b4d-83b9-35e9f1b4d8f8/sensors/1 >`

This addon has the ability to create scenes and use them in video / audio actions (start playback / stop playback). Creating a new scene corrupts the emulated_hue.json file:

`2021-01-24 22:13:13,887 ERROR aiohttp.server -- Error handling request Traceback (most recent call last): File "/usr/local/lib/python3.8/site-packages/aiohttp/web_protocol.py", line 422, in _handle_request resp = await self._request_handler(request) File "/usr/local/lib/python3.8/site-packages/aiohttp/web_app.py", line 499, in _handle resp = await handler(request) File "/app/emulated_hue/api.py", line 90, in wrapped_func return await func(cls, request) File "/app/emulated_hue/api.py", line 489, in get_full_state "scenes": await self.async_scene_to_full_state(), File "/app/emulated_hue/api.py", line 474, in async_scene_to_full_state scenes_group = scene_data["group"] KeyError: 'group'`

There is also an Ambilight option. After turning it on, I can pickup my lights, but they do not respond to the action on the screen. But I'm not sure if it's a hardware or compatibility issue...

**Reply by alexyao2015** [reply]
The first issue is that sensors aren't supported yet which is #50. The second, I'm not sure because you didn't send the request log.

**Comment by oleksandr-belei** [comment]
The second error occurred while the official HUE app is trying to connect to the Emulated HUE (after creating a scene in the Kodi addon). Here is part of emulated_hue.json
![Screenshot_20210125-002757_Microsoft Remote Desktop](https://user-images.githubusercontent.com/55722445/105645736-44fcd580-5ea4-11eb-989f-fc0a9ed0879d.jpg)

**Reply by alexyao2015** [reply]
Well seems like it created a scene without a group. I'm not sure how the official bridge handles this (does it simply add group 0 to it or what?). If you are able to either get an official bridge and provide the responses from it from `ip/api/username` and `ip/api/username/scenes` that would help in figuring out how to fix the problem.

**Reply by oleksandr-belei** [reply]
Sorry, can't help... My bridge is broken

**Reply by alexyao2015** [reply]
Sounds like we are stuck here then. Maybe you can ask the devs of that software to see? Until then, theres nothing we can do.

**Reply by oleksandr-belei** [reply]
You're right.

### Working with Groups
The Hue Bridge can support groups of lights so that you can do things like setting a colour and status to a group of lights instead of just a single light.

There is a special "All Lights" Group with an id of `0` that is defined in the bridge that a user cannot modify.

**Obtaining all Groups from the Bridge**
To obtain all the groups defined in the bridge use the **groups()** function;

```
var HueApi = require("node-hue-api").HueApi;
 
var displayResults = function(result) {
    console.log(JSON.stringify(result, null, 2));
};
 
var host = "192.168.2.129",
    username = "08a902b95915cdd9b75547cb50892dc4",
    api = new HueApi(host, username);
 
// Obtain all the defined groups in the Bridge
 
// --------------------------
// Using a promise
api.groups()
    .then(displayResults)
    .done();
 
// --------------------------
// Using a callback
api.groups(function(err, result) {
    if (err) throw err;
    displayResults(result);
});
```
This will produce an array of values detailing the id and names of the groups;

```
[
  {
    "id": "0",
    "name": "Lightset 0",
    "type": "LightGroup"
  },
  {
    "id": "1",
    "name": "VRC 1",
    "type": "LightGroup",
    "lights": [1, 2, 3, 4, 5]
  }
]
```
Please note, the **Lightset 0** group, is a special instance and will always exist and have the id of "0" as specified in the Hue Api documentation. Due to this internal group being maintained by the bridge internally, it will not return an array of light ids like the other groups in the results returned from a call to groups().

If you need to get the full details of the **Lightset 0** groups, then you can obtain that by using the getGroup() function, using an id argument of 0.

The groups function will return all types of Groups in the bridge, these include new types of groups that support the new [Hue Beyond](http://www2.meethue.com/en-us/the-range/hue-beyond).

To support the addition of these new types of groups, and the fact that most users will only want a subset of the types there are now three new functions that will filter the types of groups for you;

_**luminaires** Will obtain only the Luminaire groups (i.e. a collection of lights that make up a single device). These are not user modifiable.
**lightSources** Will obtain the Lightsource groups (i.e. a subset of the lights in a Luminarie). These are not user modifiable.
**lightGroups** Will obtain the defined groups in the bridge_

**Reply by alexyao2015** [reply]
so #62 ?

**Reply by oleksandr-belei** [reply]
It looks like that...
